### PR TITLE
Initial integration of sirius in cp2k

### PR DESCRIPTION
**Pull request by mtaillefumier**
Second try. This pull request includes the necessary files for building cp2k with sirius engine as well as tests.
They are a couple of things that were discussed in the previous pull request (that disappeared because of the rebase stuff).

- I included the missing files. gitignore was a tad too strict actually (for some directories but not others)
- did some little work on the interface but nothing groundbreaking
- SIRIUS inputs. We can do all of that with the sirius interface if wanted but it is a pain right now to export pseudo-potentials data etc... 
- SIRIUS input documentation : well todo list but it was never intended to be exposed to outside world since we always thought that the api would be used for that.
- Linear history is nearly impossible to maintain ;). it was fine with a couple of commits but does more arm than goods because we can not use branches, local clone of the official repository, etc... The list is long actually ;). Just for that commit I add to reset my local repository on github (never do that), then synchronize, and redo the same work than before. It is fine for such small commit but it will become a nightmare when you have tens of commits pilled up in your local repository.

### CODE REVIEW

**Comment by dev-zero** [comment]
to solve the formatting:

```
make pretty
make pretty
make pretty
# ... until it doesn't change anything anymore :(
git add -u
git commit --amend
git push --force
```

File: tools/toolchain/scripts/install_hdf5.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            fi
+            echo "Installing from scratch into ${pkg_install_dir}"
+            [ -d hdf5-${hdf5_ver} ] && rm -rf hdf5-${hdf5_ver}
+            tar xf hdf5-${hdf5_ver}.tar.bz2
+            cd hdf5-${hdf5_ver}
+            cd src
+
+            # this is a patch for gcc 8 since the hdf group does follow c standard.
+
+            cat <<EOF > patch.diff
```
[the remark this answers is not in the document]
Comment (mtaillefumier) [reply tools/toolchain/scripts/install_hdf5.sh]: problem solved by updating hdf5 to the latest version. The script has been cleaned up as well.

[the remark this answers is not in the document]
Comment (dev-zero) [reply tools/toolchain/scripts/install_hdf5.sh]: even better, thanks!

**Comment by dev-zero** [comment]
@mtaillefumier the current error is because you expect the directory to be `SIRIUS-5.8.0` but it is `SIRIUS-5.8` instead.

**Comment by mtaillefumier** [comment]
Ok, the last modifications seem to pass the basic tests.
- Sirius is not included in cp2k binary when MPI and openmp are not activated. 
- To check that we need to activate the parallel builds.

**Comment by oschuett** [comment]
No problem. I just triggered the pdbg test. It's a bit hidden, but you should actually be able to trigger additional tests yourself: Go a particular check run and then clicking on the "Run Test" button.

**Comment by mtaillefumier** [comment]
Hmm, it is well hidden indeed. I will look at it tomorrow. thanks for the info

**Comment by mtaillefumier** [comment]
Ok it seems I cannot trigger it the button is not present for me. My guess is that it is a test that can only be run by the admins.

**Comment by oschuett** [comment]
Uhhm, I guess one indeed needs write permissions for triggering check runs. Sorry about that.

It seems that the sirius tests are still not executed because `-D__JSON`, `-D__SIRIUS`, and possibly `-D__SPGLIB` are missing from the `DFLAGS`.

**Comment by alazzaro** [comment]
I have triggered the two tests POPT and PSMP 15 minutes ago...

**Comment by mtaillefumier** [comment]
Thanks. SIRIUS tests are not run because cp2k does not indicate that the sirius support is activated. This probably means that we forget to initialize something in the intereface.

File: tools/toolchain/scripts/install_json_fortran.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    JSON_LIBS="-ljsonfortran"
+    if [ "$with_json_fortran" != "__SYSTEM__" ] ; then
+        cat << EOF > "${BUILDDIR}/setup_json_fortran"
+prepend_path LD_LIBRARY_PATH "$pkg_install_dir/lib"
+prepend_path LD_RUN_PATH "$pkg_install_dir/lib"
+prepend_path LIBRARY_PATH "$pkg_install_dir/lib"
+prepend_path CPATH "$pkg_install_dir/include"
+export JSON_CFLAGS="${JSON_CFLAGS}"
+export JSON_LDFLAGS="${JSON_LDFLAGS}"
+export CP_DFLAGS="\${CP_DFLAGS}"
```
Comment (oschuett) [inline tools/toolchain/scripts/install_json_fortran.sh]: This line should probably read:
```
export CP_DFLAGS="\${CP_DFLAGS} -D__JSON"
```

**Comment by mtaillefumier** [comment]
How about a `touch src/cp2k_info.F` in the makefile to restart the compilation. Anyway, I added two more years for copyright (Thanks to Joost) that were missing.

**Comment by oschuett** [comment]
Right, my plan is to `dist-clean` whenever the toolchain changes in any way. However, I'm currently traveling and can't make changes to the ci-system. But I see you found a nice workaround :-)

File: tools/toolchain/scripts/install_sirius.sh [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+EOF
+        cat "${BUILDDIR}/setup_sirius" >> $SETUPFILE
+    fi
+    cat <<EOF >> "${BUILDDIR}/setup_sirius"
+export SIRIUS_CFLAGS="-I'${pkg_install_dir}/include'"
+export SIRIUS_FFLAGS="-I'${pkg_install_dir}/include'"
+export SIRIUS_LDFLAGS="${SIRIUS_LDFLAGS}"
+export SIRIUS_LIBS="${SIRIUS_LIBS}"
+export CP_DFLAGS="\${CP_DFLAGS} IF_MPI(IF_OMP(-D__SIRIUS|)|)"
+export CP_CFLAGS="\${CP_FLAGS} IF_MPI(IF_OMP(-I'${pkg_install_dir}/include'|)|)"
```
Comment (oschuett) [inline tools/toolchain/scripts/install_sirius.sh]: Typo: `\${CP_FLAGS} ` -> `\${CP_CFLAGS} `

**Comment by dev-zero** [comment]
the error in `psmp` does not seem to be related to the SIRIUS patch, correct?

**Comment by mtaillefumier** [comment]
I do not think so.

**Comment by mtaillefumier** [comment]
It is easier for me to fix on sirius side.

**Comment by mtaillefumier** [comment]
Would it be possible to have git installed because I need it to generate the patch version.

**Comment by dev-zero** [comment]
@mtaillefumier you mean whether you can assume in the SIRIUS package script that Python is installed? Or just in the test/build Docker images?

**Comment by mtaillefumier** [comment]
I need the **git** command accessible **during compilation**, not runtime. It is used in a python to generate the version.h file.

**Comment by mtaillefumier** [comment]
I found a workaround the problem but we are switching to cmake and the problem will eventually come back. So git will be needed one way or another in the container.

**Comment by dev-zero** [comment]
I would prefer not to, what are you trying to achieve with that?

**Comment by dev-zero** [comment]
not really, you can get the current hash without Git:
* https://github.com/rpavlik/cmake-modules/blob/master/GetGitRevisionDescription.cmake
* https://github.com/cp2k/cp2k/blob/a7b5debb40c8ffee62f84c6c896bd522b4744610/tools/regtesting/do_regtest#L733-L746

**Pull request merged**